Docs: Add  `verdi process status` command to tutorial (#4508)
aiidateam/aiida-core: docs/source/intro/tutorial.rst

We can see that the ``MultiplyAddWorkChain`` is currently waiting for its *child process*, the ``ArithmeticAddCalculation``, to finish.
Check the process list again for *all* processes (You should know how by now!).
After about half a minute, all the processes should be in the ``Finished`` state.

> You can also see a nice overview of the steps the work chain with the ``verdi process status`` command:

Reviewer: ```suggestion
You can also see a nice overview of the steps of the work chain with the ``verdi process status`` command:
```
Small correction aside, I am not sure if I would phrase it this way. In the context the word "steps" is often used to refer to the steps of the outline. However, what the command mostly shows is the "call stack", or the hierarchy of process calls. Now this term is probably too technical too use, but we could find a better simpler explanation. Note that the outline steps do enter in the output, but they are in between the square brackets at the end of each line. It shows the current step of each work chain, or the last step, in the case the work chain has finished.

Author: How about: "You can also see a nice overview of the processes called by the work chain with the `verdi process status` command:"? And then explain after the code snippet that the square brackets at the end of each work chain line represents the step in the outline.

.. code-block:: console

    $ verdi process status 19

>     MultiplyAddWorkChain<19> Finished [0] [3:result]

Author: One question here: what does the `[3:result]` represent? I see that it prints the `stepper_state_info` by digging a bit through the code, but it isn't clear to me what the stepper state is. 😅

Reviewer: it follows the "steps through the outline" if you check the outline of the `MultiplyAddWorkChain` you will see there are three steps, the third of which is called `result`. The stepper_state_info also correctly takes while loops and if conditions into account

        ├── multiply<20> Finished [0]
        └── ArithmeticAddCalculation<22> Finished [0]

> This command can also be very useful when debugging a work chain.

Reviewer: Why? Might be good to give a bit more detail

Author: How about (also considering your other comment [above](https://github.com/aiidateam/aiida-core/pull/4508#discussion_r511509657)):

> Between the final brackets next to the `MultiplyWorkChain` you can also see the current step in the outline (`[3:result]`). This information combined with the process state can be very useful when debugging a work chain.

Reviewer 2: Yes, let's simply replace this sentence with

```
The bracket ``[3:result]`` indicates the current step in the outline of the :py:class:`~...MultiplyWorkChain` (step 3, with name `result`).
The ``process status`` is particularly useful for debugging complex WorkChains, since it helps pinpoint where a problem occurred.

We can see that the ``MultiplyAddWorkChain`` is currently waiting for its *child process*, the ``ArithmeticAddCalculation``, to finish.
Check the process list again for *all* processes (You should know how by now!).
After about half a minute, all the processes should be in the ``Finished`` state.

> You can also see a nice overview of the processes called by the work chain with the ``verdi process status`` command:

Reviewer 2: ```suggestion
The ``verdi process status`` command prints a *hierarchical* overview of the processes called by the work chain:
```

I would usually leave it up to the user to decide whether the view is nice, but that's just a matter of taste.
